### ISSUE TITLE
[Bug]: Displaying Image version for OOTB Notebook images in Project View would be useful to have

### ISSUE THREAD

**Issue by erwangranger** [opendatahub-io/odh-dashboard#1917]
### Solutions

Review the UX screenshots: https://github.com/opendatahub-io/odh-dashboard/issues/1917#issuecomment-1760687762

* Add the version under the image name on the Workbenches table
* Apply a tooltip for the contents of the version description (same that we show in the Workbench form)

---

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

Downstream version (eg. `RHODS 1.29`)

### Version

RHODS 1.32

### Current Behavior

Using the pytorch image as an example:
![image](https://github.com/opendatahub-io/odh-dashboard/assets/32554862/ff8373fb-637e-4b31-8336-c9a87b18024f)

If I create 2 new workbenches using the 2 different versions, it's really hard to tell. 
![image](https://github.com/opendatahub-io/odh-dashboard/assets/32554862/76212f72-63f1-4ca4-9fad-4a70f013cc68)

And even unfolding does not help that much: 
![image](https://github.com/opendatahub-io/odh-dashboard/assets/32554862/2f4671ae-e438-44b2-a91a-e434c4aeaeaa)

### Expected Behavior

This (highlighted in red) is the info I feel is missing:

![image](https://github.com/opendatahub-io/odh-dashboard/assets/32554862/f6d2de7f-de4e-4024-bfb0-458fd500e049)

**Comment by dpanshug** [comment]
## UX Context

Is Erwan's expected result what we want? is there anything else we want to add?
cc @xianli123

**Comment by xianli123** [comment]
Hi @dpanshug This is my first time to see this issue. I understood why Erwan wanted to add the version info. But in another issue #1520 we added some labels for different statuses. If we add both the status labels and version info, it will get a little crowded. So we should think about these 2 issues together and come up with a solution for them. Does it make sense?

**Comment by xianli123** [comment]
Hi @erwangranger @dpanshug I am not sure if the version info is important or not. But I think we can hide this piece of info if the version information does not affect the work of the workbench. As shown below, we can make the notebook image is clickable and has a popover to show the version info. This is a common interaction in RHODS UI. What do you think? Thanks!

<img width="645" alt="Screenshot 2023-10-12 at 14 22 38" src="https://github.com/opendatahub-io/odh-dashboard/assets/53931519/6785abb3-d43d-44f5-a971-b3eda9269fc6">

**Comment by andrewballantyne** [comment]
I like the flow, only concern is how important is "the current image" really important to the user. CUDA vs PyTorch is one thing... CUDA 2023.1 vs CUDA 2023.2 is probably not as important at the top level. But I agree the information should be visible.

My two cents @xianli123 @erwangranger

**Comment by erwangranger** [comment]
Information should be accessible somewhere/somehow. That is the critical piece (80%) for me. 
Whether it's better as a popup or inline is of secondary importance to me (20%). 

But if I had to make a choice, I'd say that the "deprecated"/"deleted"/"deleted" flags will hopefully not show up too often, and I still think that leaving the screen as it currently is:
<img width="849" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/32554862/66c4ca28-0755-40dc-b753-fc44aed55c2b">
is misleading to the user. 
It really looks like I'm using the same image twice. And it's not until I'll hover over both Pytorch that I'll see it's not the case. 
To me, if 2 things are not the same, they should not have the same name. (and this is why I called my workbenches in the screenshot, A and B, so that I could tell them apart easily). 

My own opinion (I like knowing which version I'm using) is not necessarily representative of our target users, so UX colleagues, this is your call to make, and I'll be fine either way.

**Comment by xianli123** [comment]
Thanks @andrewballantyne @erwangranger Yes. I was convinced. I think we can just show the version info (with the Recommended text if it is) under the notebook image name as shown below and hide the advanced info in the popover. 

<img width="569" alt="Screenshot 2023-10-13 at 10 51 44" src="https://github.com/opendatahub-io/odh-dashboard/assets/53931519/7ccbab34-6424-44a9-93ea-01885293da10">

As far as I know, the admin can disable or delete the whole notebook image instead of a specific version of the notebook image. So if the notebook image has been disabled or deleted, we can add a label next to the notebook image name to indicate the status. What do you think? Thanks!

<img width="579" alt="Screenshot 2023-10-13 at 10 53 53" src="https://github.com/opendatahub-io/odh-dashboard/assets/53931519/5aaae297-c6f9-4a37-b565-c887a337a165">

**Comment by xianli123** [comment]
I notice the screenshot that @erwangranger attached. It's about the notebook image selection. I've seen this UI in another issue(but sorry I forgot which one). 

>  ![image](https://github.com/opendatahub-io/odh-dashboard/assets/53931519/50b19446-3254-4f0c-a808-cabb63734c34)

But at first I was a little confused about choosing notebook image and version. I am not sure if we change the components from single selection to 2 dropdowns, as shown below. In this way, this part will have the same interaction as the notebook image selection in the workbench creation process. Of course, the previous design, being able to show all the options is a good point. What do you think? @andrewballantyne @erwangranger If it's worth a shot by using 2 dropdowns, I can create a new issue to update the design.

<img width="741" alt="Screenshot 2023-10-13 at 14 57 29" src="https://github.com/opendatahub-io/odh-dashboard/assets/53931519/55cbea6d-a854-4acf-b443-4f3158eab50a">

**Comment by xianli123** [comment]
This is the issue #1690  I mentioned. FYI

**Comment by erwangranger** [comment]
Sorry I used a screenshot from the Jupyter Tile. I should not have done that. 
It's just that it had a bit more info than the double drop down:
![image](https://github.com/opendatahub-io/odh-dashboard/assets/32554862/5fc07ba4-ca33-40fe-a700-fa8f17a99b97)

I don't think we need to revisit that (^) design for now:

**Comment by xianli123** [comment]
@erwangranger Ok. Let's leave it as it is. What do you think about [the solution](https://github.com/opendatahub-io/odh-dashboard/issues/1917#issuecomment-1760687762) that I shared?

**Comment by erwangranger** [comment]
Works for me. Thanks @xianli123 .

**Comment by andrewballantyne** [comment]
I have have assumed the state of the mocks to be complete @xianli123 and moved this to Dev Ready. Let me know if you're intending on spending anymore time on the mocks. I think we have what we need from a development perspective.

**Comment by xianli123** [comment]
@andrewballantyne Yes. It's ready for dev.

**Comment by harshad16** [comment]
With the current image streams:
- The build information would also be provided in the imagestream annotations:
`opendatahub.io/notebook-build-commit: <hash>`

The requirement from UI would be to display this hash on the image selected view on the dashboard.
Along with the version info.  i,e <image name> <image version> <build-info> ex( PyTorch v1.2  `e346hu8`)
![image](https://github.com/opendatahub-io/odh-dashboard/assets/14028058/0231caa5-c0ad-4b66-b8fc-106add61db88)


Notebook manifests are already loaded with these information: [here](https://github.com/opendatahub-io/notebooks/blob/271208a7754c5ed29e02aeaac981fc80d68c08de/manifests/base/jupyter-datascience-notebook-imagestream.yaml#L23)
Dev Flag for Opendatahub development with this information:
```
workbenches:
      devFlags:
        manifests:
          - contextDir: components/odh-notebook-controller/config
            sourcePath: ''
            uri: 'https://github.com/opendatahub-io/kubeflow/tarball/v1.7.0-5'
          - contextDir: components/notebook-controller/config
            sourcePath: ''
            uri: 'https://github.com/opendatahub-io/kubeflow/tarball/v1.7.0-5'
          - contextDir: manifests
            sourcePath: ''
            uri: 'https://github.com/opendatahub-io/notebooks/tarball/2023b'
      managementState: Managed
```

**Comment by andrewballantyne** [comment]
Migrated to the issue noted above.
